packages/manager/apps/dedicated/client/app/ip/ip/firewall/ip-ip-firewall.controller.js

```diff
@@ -40,69 +42,81 @@ angular
         $scope.$broadcast('paginationServerSide.reload', 'rulesTable');
       }
 
-      $scope.$on('ips.firewall.informations.reload', () => {
-        reloadRules();
-      });
+      function fetchRules(rulesCount, offset) {
+        $scope.rulesLoading = true;
 
-      $scope.loadRules = function loadRules(rulesCount, offset) {
-        if ($scope.selectedIp) {
-          $scope.rulesLoading = true;
+        IpFirewall.getFirewallRules(
+          $scope.selectedBlock,
+          $scope.selectedIp,
+          rulesCount,
+          offset,
+        )
+          .then(
+            (rules) => {
+              $scope.rules = rules;
+              let options;
 
-          IpFirewall.getFirewallRules(
-            $scope.selectedBlock,
-            $scope.selectedIp,
-            rulesCount,
-            offset,
-          )
-            .then(
-              (rules) => {
-                $scope.rules = rules;
-                let options;
+              if (
+                $scope.rules &&
+                $scope.rules.list &&
+                $scope.rules.list.results &&
+                $scope.rules.list.results.length
```

Comment:
i think we can use lodash to avoid the long path:
```suggestion
                get($scope.rules, 'list.results.length') > 0
```

What do you think @radireddy ?

Reply:
Alternative: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Not mandatory since it has been already validated. Can be handled into a separate Pull Request.